### PR TITLE
ml - apply parent account balance to price quote

### LINKED ISSUE #48: Applying parent account balance to price quote + payment
## Background
Currently there are 2 planned ways for an account balance to be credited
1. When a student unenrolls in a course, the parent's account balance will be credited with the total tuition from all the registrations. #47 
2. When a session gets extended, the sessions need to be pro-rated again and credited to their account balance. #35 

When there's credit in the account balance, it should be calculated in the price quote. 

## Development
- [x] include an additional "apply_account_balance" boolean field in the POST request body
- [x] include an additional "parent_id" integer id field in the POST request body (only used when apply_account_balance = True)
- [x] if "apply_account_balance" is true, then return the price quote that looks like:
```
{   
   sub_total: //total of all the course tuitions,
   account_balance: //total account balance
   discounts: [ //array of discounts ],
   price_adjustment: 100,
   total: // total = sub_total - account_balance - discounts
}
```

### PR DESCRIPTION
**Pull request by tigermli**
issue #48

### CODE REVIEW

File: pricing/views.py [hunk 1]
```diff
@@ -118,7 +122,13 @@ def post(self, request):
...
         ResponseDict["discount_total"] = totalDiscountVal
         ResponseDict["price_adjustment"] = price_adjustment
-        ResponseDict["total"] = sub_total-totalDiscountVal-price_adjustment      
+        ResponseDict["total"] = sub_total-totalDiscountVal-price_adjustment
+
+        # parent balance adjustment
+        if body.get("apply_account_balance", False):
+            parent = Parent.objects.get(user_id=body["parent_id"])
+            ResponseDict["account_balance"] = float(parent.balance)
+            ResponseDict["total"] -= float(parent.balance)
```
Comment (jerrylinew) [inline pricing/views.py:131]: don't we also have to subtract the value applied from the parent balance?

**Review by jerrylinew — CHANGES_REQUESTED** [review]

File: pricing/views.py (hunk 1 above)
Comment (tigermli) [reply pricing/views.py:131]: Do you mean line 131?

Comment (tigermli) [reply pricing/views.py:131]: This is price quote so we would not subtract from the parent's balance

Comment (jerrylinew) [reply pricing/views.py:131]: ah gotcha this is just the quote

**Review by jerrylinew — APPROVED** [review]

**Pull request merged**